spec-dtslint/operators/scan-spec.ts

```diff
@@ -31,9 +31,8 @@ it('should accept seed parameter of a different type', () => {
 });
 
 it('should act appropriately with no seed', () => {
-  // Because an observable of one value will just pass that value directly through the reducer,
-  // the below could be a number or a string.
-  const a = of(1, 2, 3).pipe(scan((a: any, v) => '' + v)); // $ExpectType Observable<string | number>
+  // Starting in TS 3.5, the return type is inferred from the accumulator's type if it's provided without a seed.
+  const a = of(1, 2, 3).pipe(scan((a: any, v) => '' + v)); // $ExpectType Observable<any>
```

Comment:
Maybe type the accumulator as a `string` and expect `Observable<string>`? Is there anything to be gained by using `any` here?

Reply:
I see. Do you think it would be worth including another test that explicitly types the `value` - to make sure that works, too? I assume that `value` will also be `any` here - for the reason you gave above? FWIW, in application code I'd type it like this:

```ts
(scan((a: string, v: number) => '' + v))
```

So maybe we need a test to make sure this is okay?